src/core/scope.c

```diff
@@ -194,6 +195,9 @@ static int scope_add_extras(Scope *s) {
         if (r < 0)
                 return r;
 
+        if (s->oom_policy < 0)
+                s->oom_policy = s->cgroup_context.delegate ? OOM_CONTINUE : UNIT(s)->manager->default_oom_policy;
```

Comment:
you need propagate this into `s->cgroup_context.memory_oom_group` too. The way service.c does it